slick/src/it/scala/za/co/absa/fadb/slick/FaDbPostgresProfileSuite.scala

```diff
@@ -0,0 +1,210 @@
+
+
+  test("Test query types support with actual values") {
+
+    case class InputOutput(
+                    uuid1:      UUID,               //uuid
+                    dateTime1:  LocalDate,          //date
+                    dateTime2:  LocalTime,          //time
+                    dateTime3:  LocalDateTime,      //timestamp
+                    dateTime4:  Duration,           //interval
+                    dateTime5:  ZonedDateTime,      //timestamptz
+                    dateTime6:  OffsetDateTime,     //timestamptz
+                    range1:     Range[Int],         //range
+                    lTree1:     LTree,              //ltree
+                    map1:       Map[String, String],//hstore
+                    inet1:      InetString,         //inet
+                    macaddr1:   MacAddrString       //macaddr
+                    )
+
+    class TestFunction(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+      extends DBSingleResultFunction[InputOutput, InputOutput, SlickPgEngine]
+        with SlickFunction[InputOutput, InputOutput] {
+
+      override protected def sql(values: InputOutput): SQLActionBuilder = {
+        sql"""SELECT #$selectEntry
+            FROM #$functionName(
+              ${values.uuid1},
+              ${values.dateTime1},
+              ${values.dateTime2},
+              ${values.dateTime3},
+              ${values.dateTime4},
+              ${values.dateTime5},
+              ${values.dateTime6},
+              ${values.range1},
+              ${values.lTree1},
+              ${values.map1},
+              ${values.inet1},
+              ${values.macaddr1}
+            ) #$alias;"""
+      }
+
+      override protected def slickConverter: GetResult[InputOutput] = GetResult{r => InputOutput(
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<,
+        r.<<
+      )}
+    }
+
+    class TestSchema (implicit dBEngine: SlickPgEngine) extends DBSchema("public"){
+
+      val testFunction = new TestFunction
+    }
+
+
+    val input = InputOutput(
+      UUID.randomUUID(),
+      LocalDate.now(),
+      LocalTime.now(),
+      LocalDateTime.now(),
+      Duration.ofMinutes(42),
+      ZonedDateTime.now(),
+      OffsetDateTime.now(),
+      range1 = Range(7, 13),
+      LTree(List("This", "is", "an", "LTree")),
+      Map("a" -> "Hello", "bb" -> "beautiful", "ccc" -> "world"),
+      InetString("168.0.0.1"),
+      MacAddrString("12:34:56:78:90:ab")
+    )
+    // because postgres does not fully support time zone as Java, so we need to clear it for later successful assert
+    val expected = input.copy(dateTime5 = input.dateTime5.toOffsetDateTime.toZonedDateTime)
+
+
+    val timeout = Duration.ofMinutes(1)
+    val result = Await.result(new TestSchema()(testDBEngine).testFunction(input), timeout)
```

Comment:
Why not `AsyncFunSuite` instead of awaiting? (https://www.scalatest.org/user_guide/async_testing)